specutils/core/generic.py

```diff
@@ -0,0 +1,119 @@
+from specutils.spectrum1d import Spectrum1D
+
+import numpy as np
+import logging
+
+from astropy.nddata import NDArithmeticMixin, NDSlicingMixin, NDIOMixin
+import astropy.units as u
+from astropy.wcs import WCS
+
+
+class GenericSpectrum1D(NDIOMixin, NDSlicingMixin, NDArithmeticMixin,
+                        Spectrum1D):
+    """
+    A generalized spectrum data object that implements several `NDData`
+    features such as IO, arithmetic, and slicing.
+    """
+
+    # The Spectrum1D wcs_attribute object is useless to us, let's avoid it
+    _wcs_attributes = {}
+
+    def __init__(self, data, name="", dispersion=None, dispersion_unit=None,
+                 uncertainty=None, wcs=None, *args, **kwargs):
+        super(GenericSpectrum1D, self).__init__(flux=data,
+                                                wcs=wcs,
+                                                uncertainty=uncertainty,
+                                                indexer="UNUSED",
+                                                *args, **kwargs)
+        self._dispersion = dispersion
+        self._dispersion_unit = dispersion_unit
+        self.name = name or "New Data Object"
+
+    def from_self(self, copy=True, **kwargs):
+        """
+        Create a new `GenericSpectrum1D` object using current property
+        values. Takes all the arguments a Spectrum1D expects, arguments that
+        are not included use this instance's values.
+        """
+        self_kwargs = {"data": self._data, "dispersion": self._dispersion,
+                       "unit": self.unit, "wcs": self.wcs,
+                       "uncertainty": self._uncertainty, "mask": self.mask,
+                       "meta": self.meta}
+
+        self_kwargs.update(kwargs)
+
+        return self.__class__(**self_kwargs, copy=copy)
+
+    @classmethod
+    def from_array(cls, data, *args, **kwargs):
+        return cls(name="Array Data", data=data, **args, **kwargs)
```

Comment:
Yes. We're not using the custom WCS that the `specutils` package implements, but the Astropy WCS. I also need to include the `name` attribute.